Added boss health bar and player attack damage

PaulTashima opened the pull request:
Feel free to change *anything* or reject it outright. This was mostly a test to see if I could get it working.

- Enemy health bar added to screen.
- BossHealth script added to "Enemy"
- ScratchAbilityCollision script added to Player.ScratchArea. Not sure if this was the right solution.
- HealthBar script updated to animate changes in health quicker.

Bug: While moving and attacking OnEnterTrigger (event in ScratchAbilityCollision) can occur multiple times in one activation, causing dramatic life loss for the enemy.

buzzbuzz55 committed:
Added boss health bar and player attack damage

Feel free to change *anything* or reject it outright. This was mostly a test to see if I could get it working.

- Enemy health bar added to screen.
- BossHealth script added to "Enemy"
- ScratchAbilityCollision script added to Player.ScratchArea. Not sure if this was the right solution.
- HealthBar script updated to animate changes in health quicker.

Bug: While moving and attacking OnEnterTrigger (event in ScratchAbilityCollision) can occur multiple times in one activation, causing dramatic life loss for the enemy.

rhonig22 commented on Assets/Scripts/UXControls/HealthBar.cs:
```diff
@@ -17,7 +17,13 @@ private void Update()
+            {
+                _currentVal = _targetVal;
+            }else{
+                _currentVal = _targetVal + ((_currentVal - _targetVal) / 2);
```
It might be a good idea to incorporate Time.deltaTime here, so that you can more predictably determine how long the animation takes. But good call on adding the difference < 1 check!

rhonig22 replied:
Calculating lerpSpeed for a time of .1 seconds.

rhonig22 commented on Assets/Scripts/Player/Abilities/ScratchAbilityCollision.cs:
```diff
@@ -0,0 +1,20 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerAttackCollisionHandler : MonoBehaviour
```
This is a great start, but I was thinking of doing a different solution for scratch. Maybe picking a specific time during the animation at which to deal damage, and doing an OverlapCircle call to see what is overlapping the scratch area at that time:
https://docs.unity3d.com/ScriptReference/Physics2D.OverlapCircleAll.html

I think this would also help fix the bug you are seeing where it continually does damage.

rhonig22 replied:
Moved logic into ScratchAbility

rhonig22 committed:
Updated with changes

- Kept all scratch logic in the ability file
- Use overlapBoxAll rather than trigger area for assigning damage
- Calculate lerpspeed of healthbar
